src/Forex/ForexData.cs

```diff
@@ -0,0 +1,152 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using Newtonsoft.Json;
+
+using Azure.Migrate.Export.Common;
+using Azure.Migrate.Export.HttpRequestHelper;
+using Azure.Migrate.Export.Models;
+
+namespace Azure.Migrate.Export.Forex
+{
+    public class ForexData
+    {
+        private static ForexData Instance = null;
+        private readonly UserInput UserInputObj = null;
+        private Dictionary<string, double> ExchangeRatesUSD;
+
+        private ForexData(UserInput userInputObj, Dictionary<string, double> exchangeRatesUSD)
+        {
+            UserInputObj = userInputObj;
+            ExchangeRatesUSD = exchangeRatesUSD;
+        }
+
+        private ForexData(UserInput userInputObj)
+        {
+            UserInputObj = userInputObj;
+            ExchangeRatesUSD = null;
+        }
+
+        private ForexData()
+        {
+            UserInputObj = new UserInput();
+            ExchangeRatesUSD = null;
+        }
+
+        public static ForexData GetInstance(UserInput userInputObj)

```

Comment:
There should be single contructor and GetInstance() as its a singleton class. Only one object is created for the class.